packages/frontend/src/state/lp/hooks.ts

```diff
@@ -0,0 +1,147 @@
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk'
+import { fromTokenAmount } from '@utils/calculations'
+import { useAtomValue } from 'jotai'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import BigNumber from 'bignumber.js'
+import { OSQUEETH_DECIMALS } from '@constants/index'
+import { controllerContractAtom, controllerHelperHelperContractAtom, nftManagerContractAtom, quoterContractAtom, squeethPoolContractAtom } from '../contracts/atoms'
+import useAppCallback from '@hooks/useAppCallback'
+import { addressAtom } from '../wallet/atoms'
+import { Contract } from 'web3-eth-contract'
+import { useHandleTransaction } from '../wallet/hooks'
+import { ethers } from 'ethers'
+import { useCallback } from 'react'
+import { useGetDebtAmount, useGetVault } from '../controller/hooks'
+
+/*** CONSTANTS ***/
+const COLLAT_RATIO_FLASHLOAN = 2
+const POOL_FEE = 3000
+const MAX_INT_128 = new BigNumber(2).pow(128).minus(1).toFixed(0)
+const x96 = new BigNumber(2).pow(96)
+
+/*** GETTERS ***/
+
+export const useGetPosition = () => {
+  const contract = useAtomValue(nftManagerContractAtom)
+
+  const getPosition = useCallback(
+    async (uniTokenId: number) => {
+      if (!contract) return null
+      const position = await contract.methods.positions(uniTokenId).call()
+      const { nonce, operator, token0, token1, fee, tickLower, tickUpper, liquidity, feeGrowthInside0LastX128, feeGrowthInside1LastX128, tokensOwed0, tokensOwed1 } = position
+      return {
+        nonce,
+        operator,
+        token0,
+        token1,
+        fee,
+        tickLower,
+        tickUpper,
+        liquidity,
+        feeGrowthInside0LastX128,
+        feeGrowthInside1LastX128,
+        tokensOwed0,
+        tokensOwed1
+      }
+    },
+    [contract],
+  )
+
+  return getPosition
+}
+
+export const useGetDecreaseLiquidity = () => {
+  const contract = useAtomValue(nftManagerContractAtom)
+
+  const getDecreaseLiquiduity = useCallback(
+    async (tokenId: number, liquidity: number, amount0Min: number, amount1Min: number, deadline: number) => {
+      if (!contract) return null
+      const DecreaseLiquidityParams = {
+        tokenId,
+        liquidity,
+        amount0Min,
+        amount1Min,
+        deadline,
+      }
+
+      const decreaseLiquidity = await contract.methods.decreaseLiquidity(DecreaseLiquidityParams).call()
+
+      return decreaseLiquidity
+    },
+    [contract],
+  )
+
+  return getDecreaseLiquiduity
+}
+
+export const useGetExactIn = () => {
```

Comment:
@daryakaviani Seems like a duplicate. hooks in squeethPool has this functionality

Reply:
Few differences, so sticking with this approach:

- The squeethPool version was changed for the Auto Router for `exactIn`
- The squeethPool version doesn't allow for different fee tiers (going to add functionality for 1% pool next), so the `POOL_FEE` param will be customizable.
- Lots of token math in squeethPool version (converting to and from token amount several times to get the same value), so less elegant.